app/models/article.rb

```diff
@@ -0,0 +1,11 @@
+class Article < ApplicationRecord
+  belongs_to :user
+
+  validates :title, presence: true
+  validates :content, presence: true
+  validates :category, presence: true
+
+  scope :first_three_from_each_category, -> do 
```

Comment:
Love that you wrote a scope. This is really inefficient though as it loads everything into memory and then groups it.

Reply:
my refactor doesn't load everything into scope, but it does hit the DB four times to load things correctly. I'm not in love with this solution, but it seems slightly better than loading everything into memory